### PR TITLE
Display list of default verticals on the Site Intent Question screen

### LINKED ISSUE #16052: [Site Intent Question]  Display list of default verticals on the Site Intent Question screen
### Description

Subpart of [#15998 [Site Intent Question] Display default verticals](https://github.com/wordpress-mobile/WordPress-Android/issues/15998).
A list of clickable verticals with icons is displayed on the [Site Intent Question screen](https://github.com/wordpress-mobile/WordPress-Android/issues/16051).

### Acceptance Criteria

- [ ] Default Verticals are bundled in the app (pbArwn-3Qf-p2#default-verticals)
  - [ ] The name of each vertical is localized
- [ ] List of verticals is displayed
  - [ ] A list item is displayed as such:
    - [ ] Emoji icon in a gray circle
    - [ ] Name of the item with bold font weight
  - [ ] List item tap → Navigates to "Choose a Design" screen

#### Other Considerations:

- [ ] Unit Tests are added where applicable
- [ ] List is scrollable
- [ ] Events are tracked (Vertical selected)
   - [ ] Check the [Default vertical selected testing steps](https://github.com/wordpress-mobile/WordPress-Android/pull/16043)

### Expected Flow

Flow2

### Expected Design

Android_Topic_Question2

### PR DESCRIPTION
**Pull request by mkevins**
Fixes #16052

### Description

Subpart of [#15998 [Site Intent Question] Display default verticals](https://github.com/wordpress-mobile/WordPress-Android/issues/15998).
A list of clickable verticals with icons is displayed on the [Site Intent Question screen](https://github.com/wordpress-mobile/WordPress-Android/issues/16051).

### Acceptance Criteria

- [ ] Default Verticals are bundled in the app (pbArwn-3Qf-p2#default-verticals)
  - [ ] The name of each vertical is localized :arrow_left: The strings are in an appropriate place for localization, but I'm not sure this will happen until translations are completed (possibly during the beta release / finalization phase)
- [ ] List of verticals is displayed
  - [ ] A list item is displayed as such:
    - [ ] Emoji icon in a gray circle
    - [ ] Name of the item with bold font weight
  - [ ] List item tap → Navigates to "Choose a Design" screen

#### Other Considerations:

- [ ] Unit Tests are added where applicable :arrow_left: Not sure if this is applicable for this PR (which is mostly just UI) - open to suggestions though :smiley: 
- [ ] List is scrollable
- [ ] Events are tracked (Vertical selected)
   - [ ] Check the [Default vertical selected testing steps](https://github.com/wordpress-mobile/WordPress-Android/pull/16043)

#### Note:

A few things are remaining before this is ready for <s>review</s> merging.

1. Confirm the final list of default intents to be bundled
   * This should be easy to update before shipping, so I suppose it shouldn't block review
3. Do / should these defaults have ids and slugs? (useful for tracks events)
   * This can also be deferred (but should be resolved before merging)
5. <s>Styles need to be updated for dark mode</s>
   * This has been solved :tada:
7. Though the recyclerview is not fully populated, it still scrolls - have to track that down
   * If we plan to ship before the search / API verticals are implemented, we can address this in a hard-coded way
   * If we want to ship together with the API verticals, we'll need to solve this, though I believe we can (and should) defer this until the PR which adds the search input, to ensure it behaves correctly (i.e. that the search does not scroll off-screen)

## Regression Notes
1. Potential unintended areas of impact
None, these are changes to a new screen behind a feature flag.

8. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

9. What automated tests I added (or what prevented me from doing so)
None (since these changes are for hard-coded UI values, i.e. bundled defaults).

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentViewHolder.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentViewHolder.kt]: Do we still need this comment?

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt]: **Open Question which should not influence when we merge the PR:**
Is there any pro/con to setting the layout manager programattically vs setting it from the xml with `app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"`*?

Asking because I only found recently that [it's possible to set it from XML](https://stackoverflow.com/a/35680716/4129245), and for me it seems nicer to do it from there 😄 

I quickly tested if it works the same by setting it via XML and it does 🤷‍♂️

**\*** I guess if we do it from XML we should also add `android:orientation="vertical"`

File: WordPress/src/main/res/layout/site_creation_intents_fragment.xml
Comment (ovitrif) [inline WordPress/src/main/res/layout/site_creation_intents_fragment.xml:69]: What do you think about adding `tools:listitem="@layout/site_creation_intents_item"` for the preview during editing?

<details>
<summary>Preview</summary>

Screenshot 2022-03-21 at 17 15 44
</details>

File: WordPress/src/main/res/layout/site_creation_intents_item.xml
Comment (ovitrif) [inline WordPress/src/main/res/layout/site_creation_intents_item.xml]: I think we can safely remove this now 🙇‍♂️

File: WordPress/src/main/res/values/colors.xml
Comment (ovitrif) [inline WordPress/src/main/res/values/colors.xml]: In the designs I think the value is `#E9EFF5`, which is a bit lighter shade of grey.
Should we update it?

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsViewModel.kt
Comment (ovitrif) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsViewModel.kt]: Thank you for going with a non-confusing name, great naming 🙌

**Review by ovitrif — APPROVED** [review]
Thank you for implementing the verticals list on the UI @mkevins 🙇‍♂️

The code LGTM & testing produces the expected results ✅

Regarding the default verticals, it seems we're still deciding on the final set, as seen in this internal comment: pbArwn-3Qf-p2#comment-5152

For unit testing I think we could test:
- The code in `MainViewModel.onSiteIntentSelected`

I left a few comments though nothing blocking I think 😉

**Comment by ovitrif** [comment]
> Do / should these defaults have ids and slugs? (useful for tracks events)
This can also be deferred (but should be resolved before merging)

We're still discussing (internally here: p1647878579450379/1647541059.233989-slack-C033D5RCN2Y) and deciding on the final list of "primary site intents" a.k.a. what we initially show before starting to search.

Then we'll also make the list of the slugs but in principle we're agreeing to go [with a `snake_case`](p1647878434076929/1647541059.233989-slack-C033D5RCN2Y) representation for each intent name.

<details>
<summary>Preview</summary>

```csv
art,🎨,Art
automotive,🚗,Automotive
beauty,💅,Beauty
business_services,💼,Business & Services
community_nonprofit,🤝,Community & Non-Profit
education,🏫,Education
diy,🔨,DIY
fashion,👠,Fashion
finance,💰,Finance
film_television,🎥,Film & Television
fitness_exercise,💪,Fitness & Exercise
food,🍔,Food
gaming,🎮,Gaming
health,❤️,Health
interior_design,🛋️,Interior Design
lifestyle,🏖️,Lifestyle
local_services,ℹ️,Local Services
music,🎵,Music
news,🗞️,News
parenting,👶,Parenting
personal,📓,Personal
photography,📷,Photography
politics,🗳️,Politics
sports,⚽,Sports
travel,✈️,Travel
```
</details>

If you say this **should be resolved before merging** I hope we're going to have a decision soon (today) 🤞🏽

**Comment by ovitrif** [comment]
There are few detekt lint issues reported by the CI at the moment:

```log
[2022-03-21T16:49:20Z] > Task :WordPress:detekt FAILED
--
  | [2022-03-21T16:49:20Z] exceptions - 20min debt
  | [2022-03-21T16:49:20Z]         TooGenericExceptionThrown - [initializeFromResources] at /workdir/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsViewModel.kt:67:13
  | [2022-03-21T16:49:20Z] formatting - 5min debt
  | [2022-03-21T16:49:20Z]         NoUnusedImports - [] at /workdir/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentViewHolder.kt:5:1
  | [2022-03-21T16:49:20Z] naming - 5min debt
  | [2022-03-21T16:49:20Z]         TopLevelPropertyNaming - [defaultSuggestionItemViewType] at /workdir/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsAdapter.kt:11:19
  | [2022-03-21T16:49:20Z]
  | [2022-03-21T16:49:20Z] Overall debt: 30min
```

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentViewHolder.kt
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentViewHolder.kt]: Good question! I left this in but forgot to leave a comment. I meant to ask whether we should follow a fail-fast approach like this - I guess it makes sense, since I can't envision an item where tapping should not invoke a callback. I'll remove the comment, and add the similar fail-fast assertion. :+1:

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt
Comment (mkevins) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/SiteCreationIntentsFragment.kt]: Ah, good call! I'll use the declarative approach. :+1:

File: WordPress/src/main/res/layout/site_creation_intents_fragment.xml
Comment (mkevins) [reply WordPress/src/main/res/layout/site_creation_intents_fragment.xml:69]: Perfect! I was actually looking for that, but my autocomplete was acting funky at the time, and I didn't find it. I similarly added the tools:background colors (especially useful for dark mode styles :smile: ).

File: WordPress/src/main/res/values/colors.xml
Comment (mkevins) [reply WordPress/src/main/res/values/colors.xml]: Sure :+1: . I made a "rough guess" for these values - do you know if the dark-mode value is correct? I didn't see these specified anywhere :thinking:

Comment (mkevins) [reply WordPress/src/main/res/values/colors.xml]: I wonder whether these should be specified as standard variables :thinking: . Originally (locally), I chose standard colors:

Light: `@color/gray_5` == `#dcdcde`
Dark: `@color/gray_70` == `#3c434a`

But these weren't neutral (especially the dark color), which stood out against the container background with some significant chromatic contrast (i.e. the circle background appears blue-ish).

Container background light: `#ffffff`
Container background dark: `#2d2d2d`

Also, the next value available for lighter gray was `@color/gray_0` which is: `#f6f7f7`, which seemed too light. So, somewhat arbitrarily, I picked `#3d3d3d` for dark and a slightly more neutral `#dcdcdc` for light.

It would be good to get exact values, though. @kyleaparker do you have specific values for these in mind (for dark and light modes)?

File: WordPress/src/main/res/values/site_creation_intents.xml
Comment (mkevins) [inline WordPress/src/main/res/values/site_creation_intents.xml:32]: I'm not a big fan of this pattern, having 3 "parallel arrays", since there is a possibility they can get out-of-sync. However, this seems like a reasonable approach for now, since this allows the strings for the intent descriptions to be translated in the usual way, and the number of verticals is very small. Wdyt?

File: WordPress/src/main/res/values/colors.xml
Comment (ovitrif) [reply WordPress/src/main/res/values/colors.xml]: > do you know if the dark-mode value is correct? I didn't see these specified anywhere 🤔

Indeed, we don't have it specified AFAICT but I suppose we can go with a value that is somehow similar in terms of contrast compared to the background color on the list.

The value for the dark-mode looks good, it's of a similar contrast seen through my eyes 👀 :

<details>
<summary>Preview</summary>
<br>

</details>

We can ask @kyleaparker for feedback on the icon bg in dark mode (preview above), and adjust it afterwards; I wouldn't block the PR for the color though.

> I wonder whether these should be specified as standard variables.

I'd prefer we do that. Apparently color palette (aka standard colors) comes from [color studio](https://github.com/Automattic/color-studio).

**Comment by mkevins** [comment]
> For unit testing I think we could test:
> - The code in `MainViewModel.onSiteIntentSelected`

Good idea! Added w/ https://github.com/wordpress-mobile/WordPress-Android/pull/16126/commits/6b9173814523691b68e04916e96cb8b44f1f2f67 :+1:

File: WordPress/src/main/res/values/site_creation_intents.xml
Comment (ovitrif) [reply WordPress/src/main/res/values/site_creation_intents.xml:32]: > I'm not a big fan of this pattern, having 3 "parallel arrays", since there is a possibility they can get out-of-sync.

I'm sharing the same opinion but I think the solution serves our needs best; I guess it's always a trade-off in the sense that it's more about what we're trying to achieve than "guidelines" or "best practices" 😌

File: WordPress/src/test/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVMTest.kt
Comment (ovitrif) [inline WordPress/src/test/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVMTest.kt:113]: Thank you for adding this unit test Matthew 🙇‍♂️

File: WordPress/src/test/java/org/wordpress/android/ui/sitecreation/previews/SitePreviewViewModelTest.kt
Comment (ovitrif) [inline WordPress/src/test/java/org/wordpress/android/ui/sitecreation/previews/SitePreviewViewModelTest.kt:56]: Good improvement, thanks for adding it 🙌

**Review by ovitrif — APPROVED** [review]
Thank you for addressing the comments, the code LGTM 🎉

While testing the latest updates I got the expected results ✅, the UI is on point and the added slugs for intents get propagated correctly to the analytics tracking:

```
🔵 Tracked: enhanced_site_creation_intent_question_vertical_selected, Properties: {"vertical_id":"Business & Services","vertical_slug":"business_services"}
```

I think we're getting ready with this and it keeps getting better 🙌

**Comment by ovitrif** [comment]
I'll merge this since we tackled all the points that were mandatory to be resolved before the merge 🎉

Thank you @mkevins for the work 🙇‍♂️

**Pull request merged**
